src/main/haskell/kore/src/Kore/AST/Lens.hs

```diff
@@ -0,0 +1,304 @@
+            , equalsResultSort
+            , equalsFirst
+            , equalsSecond
+            }
+      =
+        Equals
+            <$> lensOperandSort equalsOperandSort
+            <*> lensResultSort equalsResultSort
+            <*> lensChild equalsFirst
+            <*> lensChild equalsSecond
+
+    patternLensExists Exists { existsSort, existsVariable, existsChild } =
+        Exists
+            <$> lensResultSort existsSort
+            <*> lensVariable existsVariable
+            <*> lensChild existsChild
+
+    patternLensFloor Floor { floorOperandSort, floorResultSort, floorChild } =
+        Floor
+            <$> lensOperandSort floorOperandSort
+            <*> lensResultSort floorResultSort
+            <*> lensChild floorChild
+
+    patternLensForall Forall { forallSort, forallVariable, forallChild } =
+        Forall
+            <$> lensResultSort forallSort
+            <*> lensVariable forallVariable
+            <*> lensChild forallChild
+
+    patternLensIff Iff { iffSort, iffFirst, iffSecond } =
+        Iff
+            <$> lensResultSort iffSort
+            <*> lensChild iffFirst
+            <*> lensChild iffSecond
+
+    patternLensImplies Implies { impliesSort, impliesFirst, impliesSecond } =
+        Implies
+            <$> lensResultSort impliesSort
+            <*> lensChild impliesFirst
+            <*> lensChild impliesSecond
+
+    patternLensIn
+        In
+            { inOperandSort
+            , inResultSort
+            , inContainedChild
+            , inContainingChild
+            }
+      =
+        In
+            <$> lensOperandSort inOperandSort
+            <*> lensResultSort inResultSort
+            <*> lensChild inContainedChild
+            <*> lensChild inContainingChild
+
+    patternLensNext Next { nextSort, nextChild } =
+        Next
+            <$> lensResultSort nextSort
+            <*> lensChild nextChild
+
+    patternLensNot Not { notSort, notChild } =
+        Not
+            <$> lensResultSort notSort
+            <*> lensChild notChild
+
+    patternLensOr Or { orSort, orFirst, orSecond } =
+        Or
+            <$> lensResultSort orSort
+            <*> lensChild orFirst
+            <*> lensChild orSecond
+
+    patternLensRewrites
+        Rewrites
+            { rewritesSort
+            , rewritesFirst
+            , rewritesSecond
+            }
+      =
+        Rewrites
+            <$> lensResultSort rewritesSort
+            <*> lensChild rewritesFirst
+            <*> lensChild rewritesSecond
+
+    patternLensTop Top { topSort } =
+        Top <$> lensResultSort topSort
+
+    patternLensApplication
+        Application
+            { applicationSymbolOrAlias
+            , applicationChildren
+            }
+      =
+        Application applicationSymbolOrAlias
+            <$> traverse lensChild applicationChildren
+
+-- | The sort returned by a top level constructor.
+-- NOTE ABOUT NOTATION:
+-- In the this haskell code, this is always `s2`.
```

Comment:
I'm going to remove this note because it is obsolete: the notation it documents was used in `Kore.ASTUtils.SmartConstructors`, but not anymore in this module.